Tests/FirebaseTests/Tests/FirebaseRequestTests.swift

```diff
@@ -129,22 +129,60 @@ final class FirebaseRequestTests: XCTestCase {
         XCTAssertEqual(call.body.convertToHTTPBody().data, HTTPBody(string: expectedBody).data)
     }
     
-    func test_firebaseRequest_should_returnError_when_callFailed() throws {
+    func test_firebaseRequest_should_parseFirebaseError_when_callFailed() throws {
         // Given
-        let clientStub = ClientStub(withError: FirebaseError(error: "Error during API call"))
+        let error = """
+                            {
+                                "error": {
+                                    "code": 0,
+                                    "message": "Unknown error",
+                                    "status": "UNKNOWN"
+                                }
+                            }
+                            """
+        let clientStub = ClientStub(withError: error, withStatus: HTTPStatus.notFound)
         let firebaseApiRequest = FirebaseAPIRequest.dummy(client: clientStub)
         let fakeRoute = FakeRoute(request: firebaseApiRequest)
-        let expectedBody = String.random(length: 10)
         
         // When
-        let result = try fakeRoute.fake(body: expectedBody)
+        let result = try fakeRoute.fake()
         
         // Then
-        let _ = result.catch { error in
-            XCTAssertEqual(error.localizedDescription, "⚠️ [FirebaseError.Error during API call: Error during API call]")
-        }.do { _ in
-            XCTFail("No error raised")
-        }
+        let _ = result
+            .catch { error in
+                if let firebaseError = error as? FirebaseError {
+                    XCTAssertEqual(firebaseError.identifier, String(FirebaseErrorCode.unknown.rawValue))
+                    XCTAssertEqual(firebaseError.reason, "Unknown error")
+                    XCTAssertEqual(firebaseError.error.status, "UNKNOWN")
```

Comment:
You should store these hardcoded strings into variables and then use the variable here and some string interpolation in the JSON stub above to inject them for stronger tests

Reply:
Yes, I split the test because I have a test to validate the api return and another to validate return content: 29a7d01